bayesian/dependency_finder.py

```diff
@@ -46,6 +48,14 @@ def execute(self, arguments, db, manifests):
         # If we receive a manifest file we need to save it first
         result = []
         for manifest in manifests:
+            resolve_poms = False
+            if source == 'osio':
+                content_hash = generate_content_hash(manifest['content'])
+                if pom_hash.get(content_hash) is not None:
+                    return {'result': pom_hash.get(content_hash)}
```

Comment:
this early return will not work for multiple manifests

Reply:
good catch